urllib3/_collections.py

```diff
@@ -148,11 +149,18 @@ def __init__(self, headers=None, **kwargs):
             self.extend(kwargs)
 
     def __setitem__(self, key, val):
-        return _dict_setitem(self, key.lower(), (key, val))
+        if key in self:
+            del self[key]
+        self.add(key, val)
+        return val
 
     def __getitem__(self, key):
-        val = _dict_getitem(self, key.lower())
-        return ', '.join(val[1:])
+        key_lower = key.lower()
+        _, values = _dict_getitem(self, key_lower)
+        if key_lower in SPECIAL_CASE_MULTIPLE_HEADERS:
+            # NOTE(sigmavirus24): Should we return something else?
```

Comment:
I think a list is the wrong thing here. We can't join the header values like we do elsewhere.